spring-integration-core/src/main/java/org/springframework/integration/endpoint/SourcePollingChannelAdapter.java

```diff
@@ -212,10 +215,17 @@ protected void handleMessage(Message<?> message) {
 		if (this.shouldTrack) {
 			message = MessageHistory.write(message, this, this.getMessageBuilderFactory());
 		}
+		AcknowledgmentCallback callback = StaticMessageHeaderAccessor.getAcknowledgmentCallback(message);
 		try {
 			this.messagingTemplate.send(getOutputChannel(), message);
+			if (callback != null && !callback.isAcknowledged()) {
+				callback.acknowledge(Status.ACCEPT);
```

Comment:
It's a bit speculative to always do this unconditionally.
The `this.messagingTemplate.send()` might be to the thread hands off `MessageChannel` and end-user still would like to ack in his/her code downstream somewhere. But if we do this way we deprive the general purpose of the `AcknowledgmentCallback`.
Any thoughts?

Reply:
That's a good point; however, at least with Kafka, that won't work because the consumer is not thread-safe; need to think.